TST: split up tests/plotting/test_frame.py into subdir & modules #34769

Mikhaylov-yv opened the pull request:
Created instead of #37538

- [x] closes #34769
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

I moved the file to a new directory.
What is the best structure for this folder?

[before_test_ploting_frame](https://gist.github.com/Mikhaylov-yv/999de1451e8ecc836215b30db34ef841)
[after_test_ploting_frame](https://gist.github.com/Mikhaylov-yv/98eca32c21f59cc9e7653cd96d5fabd2)

Mikhaylov-yv committed:
Moving the file test_frame.py to a new directory

Mikhaylov-yv committed:
Moving the file test_frame.py to a new directory

Mikhaylov-yv committed:
Сreated file test_frame_color.py

Mikhaylov-yv committed:
Fix merge error

Mikhaylov-yv committed:
Transfer tests
of test_frame.py
to test_frame_color.py

Mikhaylov-yv committed:
Transfer tests
of test_frame.py
to test_frame_color.py

Mikhaylov-yv committed:
Transfer tests
of test_frame.py
to test_frame_color.py

Mikhaylov-yv committed:
Transfer tests
of test_frame.py
to test_frame_color.py

Mikhaylov-yv committed:
Transfer tests
of test_frame.py
to test_frame_color.py

Mikhaylov-yv committed:
PEP 8 fixes

Mikhaylov-yv committed:
PEP 8 fixes

Mikhaylov-yv committed:
Fix merge error

Mikhaylov-yv committed:
Transfer tests

of test_frame.py
to test_frame_groupby.py and test_frame_subplots.py

Mikhaylov-yv committed:
Removing unnecessary imports

Mikhaylov-yv committed:
Removing unnecessary imports

Mikhaylov-yv committed:
Merge branch 'master' into #34769_TST_split_up_tests

# Conflicts:
#	pandas/tests/plotting/frame/test_frame.py

jbrockmendel committed:
catch FutureWarnings (#37587)

Mikhaylov-yv committed:
PEP 8 fixes

Mikhaylov-yv committed:
PEP 8 fixes

jbrockmendel committed:
TST/REF: collect indexing tests by method (#37590)

jbrockmendel committed:
TST/REF: collect indexing tests by method (#37590)

jbrockmendel committed:
TST/REF: collect indexing tests by method (#37590)

jbrockmendel committed:
REF: prelims for single-path setitem_with_indexer (#37588)

jbrockmendel committed:
ENH: __repr__ for 2D DTA/TDA (#37164)

jbrockmendel committed:
CLN: de-duplicate _validate_where_value with _validate_setitem_value (#37595)

jbrockmendel committed:
TST/REF: collect tests by method (#37589)

* TST/REF: move remaining setitem tests from test_timeseries

* TST/REF: rehome test_timezones test

* move misplaced arithmetic test

* collect tests by method

* move misplaced file

jbrockmendel committed:
REF: Categorical.is_dtype_equal -> categories_match_up_to_permutation (#37545)

MarcoGorelli committed:
CLN refactor non-core (#37580)

MarcoGorelli committed:
refactor core/computation (#37585)

jbrockmendel committed:
TST/REF: share method tests between DataFrame and Series (#37596)

jbrockmendel committed:
BUG: Index.where casting ints to str (#37591)

jbrockmendel committed:
REF: IntervalArray comparisons (#37124)

PCerles committed:
regression fix for merging DF with datetime index with empty DF (#36897)

jorisvandenbossche committed:
ERR: fix error message in Period for invalid frequency (#37602)

jbrockmendel committed:
CLN: remove rebox_native (#37608)

jbrockmendel committed:
TST/REF: tests.generic (#37618)

jbrockmendel committed:
TST: collect tests by method (#37617)

* TST/REF: collect test_timeseries tests by method

* misplaced DataFrame.values tst

* misplaced dataframe.values test

* collect test by method

jbrockmendel committed:
TST/REF: share tests across Series/DataFrame (#37616)

ssche committed:
Gh 36562 typeerror comparison not supported between float and str (#37096)

MicaelJarniac committed:
docs: fix punctuation (#37612)

arw2019 committed:
REGR: pd.to_hdf(..., dropna=True) not dropping missing rows (#37564)

jbrockmendel committed:
parametrize set_axis tests (#37619)

ivanovmg committed:
CLN: clean color selection in _matplotlib/style (#37203)

erfannariman committed:
DEPR: DataFrame/Series.slice_shift (#37601)

jbrockmendel committed:
REF: re-use validate_setitem_value in Categorical.fillna (#37597)

fangchenli committed:
PERF: release gil for ewma_time (#37389)

phofl committed:
BUG: Groupy dropped nan groups from result when grouping over single column (#36842)

attack68 committed:
ENH: implement timeszones support for read_json(orient='table') and astype() from 'object' (#35973)

twoertwein committed:
REF/BUG/TYP: read_csv shouldn't close user-provided file handles (#36997)

* BUG/REF: read_csv shouldn't close user-provided file handles

* get_handle: typing, returns is_wrapped, use dataclass, and make sure that all created handlers are returned

* remove unused imports

* added IOHandleArgs.close

* added IOArgs.close

* mostly comments

* move memory_map from TextReader to CParserWrapper

* moved IOArgs and IOHandles

* more comments

Co-authored-by: Jeff Reback <jeff@reback.net>

MarcoGorelli committed:
more typing checks to pre-commit (#37539)

ivanovmg committed:
TST: 32bit dtype compat test_groupby_dropna (#37623)

Japanuspus committed:
BUG: Metadata propagation for groupby iterator (#37461)

Mikhaylov-yv committed:
Fixed class name

Mikhaylov-yv committed:
Fixed class name

jbrockmendel committed:
BUG: read-only values in cython funcs (#37613)

Mikhaylov-yv committed:
Transfer tests

of test_frame.py
to test_frame_subplots.py

Mikhaylov-yv committed:
Transfer tests

of test_frame.py
to test_frame_subplots.py

MarcoGorelli committed:
CLN refactor core/arrays (#37581)

abbywh committed:
Fixed Metadata Propogation in DataFrame (#37381)

arw2019 committed:
TYP: add Shape alias to pandas._typing (#37128)

MicaelJarniac committed:
DOC: Fix typo (#37630)

ivanovmg committed:
CLN: parametrize test_nat_comparisons (#37195)

taytzehao committed:
dataframe dataclass docstring updated (#37632)

MarcoGorelli committed:
refactor core/groupby (#37583)

ma3da committed:
BUG: set index of DataFrame.apply(f) when f returns dict (#37544) (#37606)

arw2019 committed:
BUG: to_dict should return a native datetime object for NumPy backed dataframes (#37571)

twoertwein committed:
ENH: memory_map for compressed files (#37621)

junjunjunk committed:
DOC: add example & prose of slicing with labels when index has duplicate labels  (#36814)

* DOC: add example & prose of slicing with labels when index has duplicate labels #36251

* DOC: proofread the sentence.

Co-authored-by: Jun Kudo <jun-lab@junnoMacBook-Pro.local>

MicaelJarniac committed:
DOC: Fix typo (#37636)

"columns(s)" sounded odd, I believe it was supposed to be just "column(s)".

Mikhaylov-yv committed:
Transfer tests

of test_frame.py
to test_frame_groupby.py, test_frame_subplots.py, test_frame_color.py

Mikhaylov-yv committed:
Transfer tests

of test_frame.py
to test_frame_groupby.py, test_frame_subplots.py, test_frame_color.py

Mikhaylov-yv committed:
Changed class names

Mikhaylov-yv committed:
Changed class names

Mikhaylov-yv committed:
Removed unnecessary imports

Mikhaylov-yv committed:
Removed unnecessary imports

jbrockmendel committed:
CI: troubleshoot win py38 builds (#37652)

jbrockmendel committed:
TST/REF: collect indexing tests by method (#37638)

jbrockmendel committed:
TST/REF: collect tests for get_numeric_data (#37634)

* misplaced loc test

* TST/REF: collect get_numeric_data tests

jbrockmendel committed:
REF: de-duplicate _validate_insert_value with _validate_scalar (#37640)

jreback commented:
in pre-commit:

```pandas/tests/plotting/frame/test_frame_groupby.py:7:1: F401 'pandas as pd' imported but unused```

also merge master and ping on green.

Mikhaylov-yv committed:
Removed import

Mikhaylov-yv committed:
Removed import

Mikhaylov-yv commented:
I am very sorry. I don’t understand what to do next?

jbrockmendel committed:
CI: catch windows py38 OSError (#37659)

jbrockmendel committed:
share test (#37679)

ivanovmg committed:
TST: match matplotlib warning message (#37666)

* TST: match matplotlib warning message

* TST: match full message

ivanovmg committed:
TST: match matplotlib warning message (#37666)

* TST: match matplotlib warning message

* TST: match full message

phofl committed:
pd.Series.loc.__getitem__ promotes to float64 instead of raising KeyError (#37687)

jbrockmendel commented:
> I am very sorry. I don’t understand what to do next?

run `git pull upstream master` to merge master, then in the new files use `from pandas import foo` instead of `pd.foo`

jbrockmendel committed:
REF/TST: misplaced Categorical tests (#37678)

jbrockmendel committed:
REF/TST: collect indexing tests by method (#37677)

jbrockmendel committed:
CLN: only call _wrap_results one place in nanmedian (#37673)

jbrockmendel committed:
TYP: Index._concat (#37671)

jbrockmendel committed:
BUG: CategoricalIndex.equals casting non-categories to np.nan (#37667)

jbrockmendel committed:
CLN: _replace_single (#37683)

* REF: simplify _replace_single by noting regex kwarg is bool

* Annotate

* CLN: remove never-False convert kwarg

jbrockmendel committed:
TYP: make more internal funcs keyword-only (#37688)

jbrockmendel committed:
REF: make Series._replace_single a regular method (#37691)

ivanovmg committed:
REF: simplify cycling through colors (#37664)

jbrockmendel committed:
REF: implement _wrap_reduction_result (#37660)

AlexKirko committed:
BUG: preserve fold in Timestamp.replace (#37644)

phofl committed:
CLN: Clean indexing tests (#37689)

ivanovmg committed:
TST: fix warning for pie chart (#37669)

ivanovmg committed:
TST: fix warning for pie chart (#37669)

ukarroum committed:
PERF: reverted change from commit 7d257c6 to solve issue #37081 (#37426)

charlesdong1991 committed:
DataFrameGroupby.boxplot fails when subplots=False (#28102)

phofl committed:
ENH: Improve error reporting for wrong merge cols (#37547)

Mikhaylov-yv commented:
Thanks @jbrockmendel. After that, I get a message like this:
![image](https://user-images.githubusercontent.com/44315225/98464889-e8f4b000-21d6-11eb-9f1b-0722a8f7219e.png)
Am I doing the right thing.

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

pep8speaks commented:
Hello @Mikhaylov-yv! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`pandas/tests/plotting/frame/test_frame_color.py`](https://github.com/pandas-dev/pandas/blob/78f42041522b2beb05764cdfb53742d6647ea521/pandas/tests/plotting/frame/test_frame_color.py):

> [Line 52:5](https://github.com/pandas-dev/pandas/blob/78f42041522b2beb05764cdfb53742d6647ea521/pandas/tests/plotting/frame/test_frame_color.py#L52): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 633:58](https://github.com/pandas-dev/pandas/blob/78f42041522b2beb05764cdfb53742d6647ea521/pandas/tests/plotting/frame/test_frame_color.py#L633): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

Mikhaylov-yv committed:
PEP8

Mikhaylov-yv committed:
PEP8

Mikhaylov-yv committed:
PEP8

Mikhaylov-yv committed:
PEP8

jbrockmendel commented:
> Am I doing the right thing?

I dont see anything wrong in the screenshot you posted.

Near the bottom of this page you should see a box with red text saying "Some checks were not successful".  In that box there are two red "x"s for the failed checks.  Click through to the details on those to see what still needs to be addressed.

jbrockmendel commented on pandas/tests/plotting/frame/test_frame_color.py:
```diff
@@ -0,0 +1,632 @@
+        tm.close()
+
+    def test_bar_user_colors(self):
+        df = pd.DataFrame(
```
the linter will want this to be just DataFrame

MarcoGorelli committed:
Move inconsistent namespace check to pre-commit, fixup more files (#37662)

* check for inconsistent namespace usage

* doc

* typos

* verbose regex

* use verbose flag

* use verbose flag

* match both directions

* add test

* don't import annotations from future

* update extra couple of cases

* :truck: rename

* typing

* don't use subprocess

* don't type tests

* use pathlib

jbrockmendel committed:
REF: simplify NDFrame.replace, ObjectBlock.replace (#37704)

jbrockmendel committed:
REF: implement Categorical.encode_with_my_categories (#37650)

* REF: implement Categorical.encode_with_my_categories

* privatize

jbrockmendel committed:
BUG: unpickling modifies Block.ndim (#37657)

jbrockmendel committed:
REF: dont support dt64tz in nanmean (#37658)

rhshadrach committed:
CLN: Simplify groupby head/tail tests (#37702)

phofl committed:
Bug in loc raised for numeric label even when label is in Index (#37675)

jbrockmendel committed:
REF: implement replace_regex, remove unreachable branch in ObjectBlock.replace (#37696)

Mikhaylov-yv committed:
Fixes for linter

Mikhaylov-yv committed:
Fixes for linter

simonjayhawkins committed:
TYP: Check untyped defs (except vendored) (#37556)

jbrockmendel committed:
REF: remove ObjectBlock._replace_single (#37710)

Mikhaylov-yv committed:
Сhange pd.DateFrame to DateFrame

Mikhaylov-yv committed:
Сhange pd.DateFrame to DateFrame

Mikhaylov-yv commented:
@jbrockmendel Thanks. I think I figured out what to do.

Mikhaylov-yv committed:
Сhange DateFrame to pd.DateFrame

Mikhaylov-yv committed:
Сhange DateFrame to pd.DateFrame

Mikhaylov-yv committed:
Сhange pd.DateFrame to DateFrame

Mikhaylov-yv committed:
Сhange pd.DateFrame to DateFrame

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

Mikhaylov-yv committed:
Removing imports

Mikhaylov-yv committed:
Removing imports

Mikhaylov-yv committed:
Bug fixes

Mikhaylov-yv committed:
Bug fixes

Mikhaylov-yv committed:
Bug fixes

Mikhaylov-yv committed:
Bug fixes

Mikhaylov-yv commented:
Can you tell me what to do with this error message? This import is used in the **pandas / _testing.py** file

> Check for non-standard numpy.random-related imports excluding pandas/_testing.py.......................Failed
> - hook id: non-standard-numpy.random-related-imports
> - exit code: 1
> 
> pandas/tests/plotting/frame/test_frame.py:9:from numpy.random import rand, randn
> pandas/tests/plotting/frame/test_frame_color.py:6:from numpy.random import rand, randn

Do I need to fix **rand()** to the **np.random.rand()**

Mikhaylov-yv committed:
Fix incorrect merge

Mikhaylov-yv committed:
Fix incorrect merge

Mikhaylov-yv committed:
Merge branch 'master' into #34769_TST_split_up_tests

# Conflicts:
#	pandas/tests/plotting/frame/test_frame.py

Mikhaylov-yv committed:
test_frame_color.py edit

Mikhaylov-yv committed:
test_frame_color.py edit

Mikhaylov-yv committed:
Transfer tests

of test_frame.py
to test_frame_color.py, test_frame_groupby.py and test_frame_subplots.py

Mikhaylov-yv committed:
Transfer tests

of test_frame.py
to test_frame_color.py, test_frame_groupby.py and test_frame_subplots.py

Mikhaylov-yv committed:
Removing unnecessary imports

Mikhaylov-yv committed:
Removing unnecessary imports

Mikhaylov-yv committed:
PEP8

Mikhaylov-yv committed:
PEP8

Mikhaylov-yv committed:
# Conflicts:
#	pandas/tests/plotting/frame/test_frame.py
#	pandas/tests/plotting/frame/test_frame_color.py
#	pandas/tests/plotting/frame/test_frame_subplots.py

phofl committed:
CLN: clean categorical indexes tests (#37721)

Mikhaylov-yv committed:
Merge branch '#34769_TST_split_up_tests_1' into #34769_TST_split_up_tests

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

Mikhaylov-yv committed:
Merge remote-tracking branch 'origin/#34769_TST_split_up_tests' into #34769_TST_split_up_tests

Mikhaylov-yv committed:
Fix merge error

Mikhaylov-yv committed:
Merge branch '#34769_TST_split_up_tests_1' into #34769_TST_split_up_tests

# Conflicts:
#	pandas/tests/plotting/frame/test_frame.py
#	pandas/tests/plotting/frame/test_frame_color.py
#	pandas/tests/plotting/frame/test_frame_subplots.py

Mikhaylov-yv committed:
Merge remote-tracking branch 'origin/#34769_TST_split_up_tests' into #34769_TST_split_up_tests

Mikhaylov-yv committed:
Fix merge error

Mikhaylov-yv commented:
I hope I took into account all my mistakes

Mikhaylov-yv committed:
Merge branch 'master' into #34769_TST_split_up_tests

Mikhaylov-yv commented:
The errors don't seem to be relevant to my pull request. I'm wrong?

Mikhaylov-yv committed:
Removing unnecessary features

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

jbrockmendel commented:
> The errors don't seem to be relevant to my pull request. I'm wrong?

You are not wrong.  This is affecting all the builds, you can ignore this failure for now.

Mikhaylov-yv commented:
Thanks @jbrockmendel Does this mean the job is ready?  or do you need to do something else?

Mikhaylov-yv committed:
Resolving Commit Conflicts daf999f 365d843

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

# Conflicts:
#	pandas/tests/plotting/frame/test_frame.py

Mikhaylov-yv committed:
black fix

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

jbrockmendel commented:
@Mikhaylov-yv is this pretty much cut/paste?  if so, can you merge master and ping on green

Mikhaylov-yv commented:
@jbrockmendel Yes it is.  the number of tests is the same as the master.  
Now I'm going to merge

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

Mikhaylov-yv committed:
Merge remote-tracking branch 'upstream/master' into #34769_TST_split_up_tests

jbrockmendel commented:
LGTM cc @jreback

jreback commented:
thanks @Mikhaylov-yv

Mikhaylov-yv commented:
@jreback @jbrockmendel  Thank you very much for your help
